### PR TITLE
Merge CX Reusable into Master

### PR DESCRIPTION
**Pull request by ear-dev**
- New json config style for all settings
- DF.app now handles multiple DF.agent endpoints using a mapping between omnichannel agents and df_config.json instances.

### CODE REVIEW

**Comment by ear-dev** [comment]
@Shailesh351 fwiw, these are the complete set of settings I'm using so far in our dev and preprod rocketchat instances...

```
{
	"project_id": "xxxxx",
	"client_email": "xxxxx",
	"agent_id": "xxxxx",
	"agent_region": "us-central1",
	"agent_default_language": "EN",
	"environment": "preprod",
	"environment_id": "8bb045b5-2589-4d4d-a4df-ab77edd4a0c9",
	"private_key": "-----BEGIN PRIVATE KEY-----\n-----END PRIVATE KEY-----\n",
	"agent_version": "CX",
	"fallback_responses_limit": 3,
	"fallback_target_department": "",
	"handover_message": "",
	"no_agents_for_handover_message": "",
	"service_unavailable_message": "",
	"close_chat_message": "",
	"hide_quickreplies": true,
	"enable_chat_closed_by_visitor_event":false,
	"chat_closed_by_visitor_event": "end_live_chat",
	"enable_welcome_message": false,
	"welcome_message": "",
	"welcome_intent_on_start": false,
	"enable_customer_timeout": false,
	"customer_timeout_time": 240,
	"customer_timeout_warning_time": 180,
	"customer_timeout_warning_message": "",
	"session_maintenance_interval": "5 minutes",
	"session_maintenance_event_name": "session_maintenance",
	"dialogflow_file_attachment_event_name": "invalidUserInput",
	"dialogflow_disable_composer_on_trigger_event": false
}
```

File: package.json [hunk 1]
```diff
@@ -1,8 +1,8 @@
 {
     "devDependencies": {
-        "@rocket.chat/apps-engine": "git+https://github.com/WideChat/Rocket.Chat.Apps-engine.git#master",
+        "@rocket.chat/apps-engine": "git+https://github.com/WideChat/Rocket.Chat.Apps-engine.git#multicode_setting_type",
```
Comment (Shailesh351) [inline package.json:3]: We have now merged `multicode_setting_type` in `master`, So we should be good with the master branch. Just make sure it has the latest commit in package-lock.json file

**Comment by Shailesh351** [comment]
@ear-dev @AlexanderKanakis I have been testing it locally and didn't find any issue so far. So we are good to merge it.

File: package.json (hunk 1 above)
Comment (ear-dev) [reply package.json:3]: @AlexanderKanakis @Shailesh351 looks like this one last change is required then we're good to go?

Comment (AlexanderKanakis) [reply package.json]: Ready.

Comment (Shailesh351) [reply package.json]: @AlexanderKanakis Can you update the package-lock.json file as well

Comment (Shailesh351) [reply package.json]: Currently the hash is `41bdb50dd83c798796ff781fff40323b52590489 ` https://github.com/WideChat/Apps.Dialogflow/blob/300ef80e6016c687b2762c2986bb13034a62bc43/package-lock.json#L32

and latest on apps-engine is `7777d4b7702500a3c2ff28aaae9fa8963dba109f`

I think we haven't updated it on the DF App repo

Comment (AlexanderKanakis) [reply package.json]: Updated apps-engine to `7777d4b7702500a3c2ff28aaae9fa8963dba109f`

Comment (Shailesh351) [reply package.json]: Thanks, @AlexanderKanakis! 🙌

**Review by Shailesh351 — APPROVED** [review]
I have been testing it locally and didn't find any issue so far.

**Comment by ear-dev** [comment]
Ok......  going in!  Thanks guys, this is a big milestone for us.

**Pull request merged**